src/components/NestedInputCheckbox.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+  <div 
+    :class="value ? 'ft-input-checkbox--checked' : 'ft-input-checkbox--unchecked'"
+    class="ft-input-checkbox"
+  >
+    <label 
+      class="ft-input-checkbox__label"
+    >
+      <input 
+        :id="id" 
+        :disabled="disabled"
+        :value="val"
+        v-model="checked"
+        class="ft-input-checkbox__native" 
+        type="checkbox"
+        v-on="listeners">
+      <transition-group name="scale">
+        <CheckboxSelectedOpen
+          v-if="isChecked && areAllChecked"
+          key="selected-open"
+          class="ft-input-checkbox__checkbox"
+        />
+        <CheckboxSelected
+          v-if="isChecked && areSomeChecked"
+          key="selected"
+          class="ft-input-checkbox__checkbox"
+        />
+        <CheckboxUnselected 
+          v-if="!isChecked"
+          key="unselected"
+          class="ft-input-checkbox__checkbox"
+        />
+      </transition-group>
+      <div
+        v-if="label" 
+        class="ft-input-checkbox__label-content">
+        <p class="ft-input-checkbox__label-content__label">{{ label }}</p>
+        <slot/>
+      </div>
+    </label>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue"
+import {CheckboxGroup} from 'types'
+
+import { 
+  CheckboxSelected24 as CheckboxSelected, 
```

Comment:
This is hard-coding the icons to the base 24px ones. Is there a chance the user will want another size here?